lib/PuppeteerSharp/NavigatorWatcher.cs

```diff
@@ -77,7 +78,7 @@ internal class NavigatorWatcher
         public Task<bool> SameDocumentNavigationTask => _sameDocumentNavigationTaskWrapper.Task;
         public Task<bool> NewDocumentNavigationTask => _newDocumentNavigationTaskWrapper.Task;
         public Response NavigationResponse => _navigationRequest?.Response;
-        public Task<Task> TimeoutOrTerminationTask => Task.WhenAny(_timeoutTask, _terminationTaskWrapper.Task);
+        public CancellationToken TimedCancellationToken => CancellationTokenSourceHelper.GetCancellationToken(_timeout);
```

Comment:
Let's call it `TimeoutCancellationToken`

Reply:
We should trigger the cancellation token here https://github.com/kblok/puppeteer-sharp/blob/a890badffa63bb8dca833e8b95005e06ed54d7ba/lib/PuppeteerSharp/NavigatorWatcher.cs#L120